ligo/src/types.mligo

```diff
@@ -0,0 +1,230 @@
+  ; pending_owner : address
+  ; migration_status : migration_status
+  ; voting_period : nat
+  ; quorum_threshold : nat
+  ; extra : contract_extra
+  ; proposals : (proposal_key, proposal) big_map
+  ; proposals_key_list_sort_by_date : proposal_key list
+  ; permits_counter : nat
+  }
+
+// -- Parameter -- //
+
+type transfer_destination =
+  [@layout:comb]
+  { to_ : address
+  ; token_id : token_id
+  ; amount : nat
+  }
+type transfer_item =
+  [@layout:comb]
+  { from_ : address
+  ; txs : transfer_destination list
+  }
+type transfer_params = transfer_item list
+
+type balance_request_item =
+  [@layout:comb]
+  { owner : address
+  ; token_id : token_id
+  }
+type balance_response_item =
+  [@layout:comb]
+  { request : balance_request_item
+  ; balance : nat
+  }
+type balance_request_params =
+  [@layout:comb]
+  { requests : balance_request_item list
+  ; callback : balance_response_item list contract
+  }
+
+type token_metedata_registry_param = address contract
+
+type operator_param =
+  [@layout:comb]
+  { owner : address
+  ; operator : address
+  ; token_id : token_id
+  }
+type update_operator =
+  [@layout:comb]
+  | Add_operator of operator_param
+  | Remove_operator of operator_param
+type update_operators_param = update_operator list
+
+type fa2_parameter =
+    Transfer of transfer_params
+  | Balance_of of balance_request_params
+  | Token_metadata_registry of token_metedata_registry_param
+  | Update_operators of update_operators_param
+
+type transfer_ownership_param = address
+
+type migrate_param = address
+
+type voting_period = nat
+type quorum_threshold = nat
+
+type propose_params =
+  { frozen_token : nat
+  ; proposal_metadata : proposal_metadata
+  }
+
+type vote_param =
+  [@layout:comb]
+  { proposal_key : proposal_key
+  ; vote_type : vote_type
+  ; vote_amount : nat
+  }
+type vote_param_permited =
+  { argument : vote_param
+  ; permit : permit option
+  }
+
+type burn_param =
+  [@layout:comb]
+  { from_ : address
+  ; token_id : token_id
+  ; amount : nat
+  }
+type mint_param =
+  [@layout:comb]
+  { to_ : address
+  ; token_id : token_id
+  ; amount : nat
+  }
+
+type token_address_param = address contract
+
+type transfer_contract_tokens_param =
+  { contract_address : address
+  ; params : transfer_params
+  }
+
+type vote_permit_counter_param =
+
+  [@layout:comb]
+  { param : unit
+  ; callback : nat contract
+  }
+
+type parameter =
+    Call_FA2 of fa2_parameter
+  | Transfer_ownership of transfer_ownership_param
+  | Accept_ownership of unit
+  | Migrate of migrate_param
+  | Confirm_migration of unit
+  | Propose of propose_params
+  | Vote of vote_param_permited list
+  | Set_voting_period of voting_period
+  | Set_quorum_threshold of quorum_threshold
+  | Flush of unit
+  | Burn of burn_param
+  | Mint of mint_param
+  | Transfer_contract_tokens of transfer_contract_tokens_param
+  | Token_address of token_address_param
+  | GetVotePermitCounter of vote_permit_counter_param
+
```

Comment:
I had written a couple of comments already but then realized this is a bit behind master and the PR needs a rebase.

Listing them here just in case they come useful, compared to the Lorentz version LIGO:
- lacks a `Drop_proposal` equivalent here in `parameter`
  (and as a consequence a corresponding entrypoint function)
- has `unit` as the argument of `Flush`, but it should actually be `nat option`

Maybe some of the comments above also are just about outdated stuff, sorry if that's the case.

Reply:
Yes, these are just the very recent additions, should automatically be handled on rebase.